lib/h3.rb

```diff
@@ -109,4 +111,18 @@ def self.hex_ring(h3_index, k)
     hexRing(h3_index, k, hexagons)
     hexagons.read_array_of_ulong_long(max_hexagons).reject { |i| i == 0 }
   end
+
+  def self.origin_h3_indexes_from_unidirectional_edge(edge)
+    max_hexagons = 10 # What is the theoretical max here?
```

Comment:
2

Reply:
Also the name of this method is a bit confusing to me, because there's an origin and destination hexagon for each unidirectional edge, but this is returning both the origin and destination.

Reply:
Examples:

```js
damocles@Talyn:~/oss/h3-node(master)$ node
> const h3 = require('.')
undefined
> (node:880) Warning: N-API is an experimental feature and could change at any time.
> h3.geoToH3(37.7739, -122.4313, 9)
'892830828a7ffff'
> h3.hexRing('892830828a7ffff', 1)
[ '892830828b7ffff',
  '892830828a3ffff',
  '892830828afffff',
  '892830829dbffff',
  '892830829d3ffff',
  '89283082d6bffff' ]
> h3.getH3UnidirectionalEdgesFromHexagon('892830828a7ffff')
[ '1192830828a7ffff',
  '1292830828a7ffff',
  '1392830828a7ffff',
  '1492830828a7ffff',
  '1592830828a7ffff',
  '1692830828a7ffff' ]
> h3.getH3UnidirectionalEdgesFromHexagon('892830828a7ffff').map(u => h3.getDestinationH3IndexFromUnidirectionalEdge(u))
[ '892830829d3ffff',
  '892830828afffff',
  '892830829dbffff',
  '892830828b7ffff',
  '89283082d6bffff',
  '892830828a3ffff' ]
> h3.hexRing('892830828a7ffff', 1).sort()
[ '892830828a3ffff',
  '892830828afffff',
  '892830828b7ffff',
  '892830829d3ffff',
  '892830829dbffff',
  '89283082d6bffff' ]
> h3.getH3UnidirectionalEdgesFromHexagon('892830828a7ffff').map(u => h3.getDestinationH3IndexFromUnidirectionalEdge(u)).sort()
[ '892830828a3ffff',
  '892830828afffff',
  '892830828b7ffff',
  '892830829d3ffff',
  '892830829dbffff',
  '89283082d6bffff' ]
>
```

The unidirectional edge indexes store a starting hexagon and the direction to traverse to an immediate neighbor, so you can also index transitions between hexagons, not just counts of things that fall within a hexagon. The top bits to represent the H3 indexes were chosen so they line up with the hexadecimal representation. If it starts with an 8 and is 15 characters long it's a regular H3 index, if it starts with a 1 and is 16 characters long its a unidirectional edge. The character that would normally house an 8 now houses the numbers 1-6, matching the 1-6 in this diagram:

![direction diagram](https://uber.github.io/h3/images/cpidigits.png?__uncache=11%2F25%2F2018%2C%208%3A48%3A49%20AM)

And a 0 or 7 are invalid digits at that position. The digit after that is the resolution, from 0 to F (0 to 15), and then all of the characters after that define the base cell and the child cells as compactly as possible. Unfortunately these could not be aligned with hexadecimal boundaries. The child cells are defined as 3-bit chunks that go from 0 to 7, with 7 not existing in that diagram and therefore being invalid and acting as an "undefined" placeholder, hence all of the f's at the end.

Anyways, I'm just hoping you'll be willing to adjust the naming of this function to improve clarity for the Ruby developers using it. :)

Reply:
Thanks @dfellis ! That's really cleared it up for me. I'm happy to rename!